src/pendulum_visualizer.py

```diff
@@ -39,23 +41,33 @@ def __init__(self):
         self.center_of_mass = self.ax.plot(0, -self.ac1, zorder=1, color='b', marker='o', markersize=14)
 
     def draw(self, context):
-        theta = self.EvalVectorInput(context, 0).get_value()[0]
+        if isinstance(context, Context):
+            theta = self.EvalVectorInput(context, 0).get_value()[0]
+        else:
+            theta = context
+
         path = self.arm[0].get_path()
         path.vertices[:,0] = self.arm_x*math.cos(theta)-self.arm_y*math.sin(theta)
         path.vertices[:,1] = self.arm_x*math.sin(theta)+self.arm_y*math.cos(theta)
         self.center_of_mass[0].set_data(self.ac1*math.sin(theta),-self.ac1*math.cos(theta))
 
-    def animation_update(self, i):
-        self.draw(math.sin(0.03*i))
-
-    def animate(self, log):
+    def animate(self, log, rate, resample=True, repeat=False):
         # log - a reference to a pydrake.systems.primitives.SignalLogger that
-        # constains the pendulum state, or pendulum output (the first element
-        # should be the pendulum's theta) after running a simulation.
+        # constains the plant state after running a simulation.
+        # rate - the frequency of frames in the resulting animation
+        # resample -- should we do a resampling operation to make
```

Comment:
Why not just always do it?

Reply:
Was a little scared it could add significant delay, which would be nice to be able to avoid when you know you've set your system up correctly. Doesn't hurt?

Reply:
Could it really add delay (compared to the actual rendering)?  Simple is good.

Reply:
Oh, I'm pretty OK with removing this. :/